test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_image_arguments.cpp

```diff
@@ -0,0 +1,432 @@
+{
+    using BasicMutableCommandBufferTest::BasicMutableCommandBufferTest;
+
+    MutableDispatchImage1DArguments(cl_device_id device, cl_context context,
+                                    cl_command_queue queue)
+        : BasicMutableCommandBufferTest(device, context, queue)
+    {}
+
+    virtual cl_int SetUp(int elements) override
+    {
+        BasicMutableCommandBufferTest::SetUp(elements);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        cl_bool image_support;
+
+        cl_int error =
+            clGetDeviceInfo(device, CL_DEVICE_IMAGE_SUPPORT,
+                            sizeof(image_support), &image_support, nullptr);
+        test_error(error, "clGetDeviceInfo for CL_DEVICE_IMAGE_SUPPORT failed");
+
+        cl_mutable_dispatch_fields_khr mutable_capabilities;
+
+        bool mutable_support =
+            !clGetDeviceInfo(
+                device, CL_DEVICE_MUTABLE_DISPATCH_CAPABILITIES_KHR,
+                sizeof(mutable_capabilities), &mutable_capabilities, nullptr)
+            && mutable_capabilities & CL_MUTABLE_DISPATCH_LOCAL_SIZE_KHR;
+
+        return (!mutable_support && !image_support)
+            || BasicMutableCommandBufferTest::Skip();
+    }
+
+    cl_int Run() override
+    {
+        const char *sample_const_arg_kernel =
+            R"(__kernel void sample_test( read_only image1d_t source, sampler_t
+            sampler, write_only image1d_t dest)
+            {
+               int offset = get_global_id(0);
+
+               int4 color = read_imagei( source, sampler, offset );
+
+               write_imagei( dest, offset, color );
+            })";
+
+        cl_int error;
+        clProgramWrapper program;
+        clKernelWrapper kernel;
+
+        cl_image_desc image_desc;
+        memset(&image_desc, 0x0, sizeof(cl_image_desc));
+        image_desc.image_type = CL_MEM_OBJECT_IMAGE1D;
+        image_desc.image_width = 4;
+        image_desc.image_row_pitch = 0;
+        image_desc.num_mip_levels = 0;
+
+        const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+
+        image_descriptor imageInfo = { 0 };
+        imageInfo.type = CL_MEM_OBJECT_IMAGE1D;
+        imageInfo.format = &formats;
+        imageInfo.width = 4;
+
+        BufferOwningPtr<char> imageValues, outputData;
+        MTdataHolder d(gRandomSeed);
+        generate_random_image_data(&imageInfo, imageValues, d);
+        generate_random_image_data(&imageInfo, outputData, d);
+
+        char *host_ptr = (char *)imageValues;
+
+        clMemWrapper src_image = create_image_1d(
+            context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, &formats,
+            image_desc.image_width, 0, host_ptr, nullptr, &error);
+        test_error(error, "create_image_1d failed");
+
+        clMemWrapper dst_image = create_image_1d(
+            context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR, &formats,
+            image_desc.image_width, 0, host_ptr, nullptr, &error);
```

Comment:
We should create `dst_image` with different underlying data than `src_image`, otherwise it's hard to know if data was to it copied at all.